config/entry.go

```diff
@@ -29,17 +30,22 @@ func makeEntryFromValue(value any) *Entry {
 		kind = t.Elem().Kind()
 		isSlice = true
 	}
-	return &Entry{value, []any{}, kind, isSlice}
+	return &Entry{value, []any{}, kind, isSlice, false}
 }
 
 func (e *Entry) validate(key string) error {
-	if e.Value == nil { // nil values means unset
+	if e.Value == nil {
```

Comment:
A simple `nil` comparison may silently fail in some scenarios. For example if the concrete type of `e.Value` is not `any`.

[Playground](https://goplay.tools/snippet/bBFLhCN0DLi)

We should check the reflect value instead:
```go
v := reflect.ValueOf(strct.Value)
ok := v.IsValid() && !v.IsNil()
```

To avoid multiplying reflect calls, let's replace the `reflect.TypeOf(e.Value)` with `reflect.ValueOf(e.Value)` and do the check just before `tryConversion()`. And remove `list := reflect.ValueOf(e.Value)` below.

Reply:
Done ✅ 
A bit different, but check it out 🙂